src/dict/stardict.cc

```diff
@@ -571,18 +571,36 @@ string StardictDictionary::handleResource( char type, char const * resource, siz
     case 'n': // WordNet data. We don't know anything about it.
       return "<div class=\"sdct_n\">" + Html::escape( string( resource, size ) ) + "</div>";
 
-    case 'r': // Resource file list. For now, only img: is handled.
+    case 'r': // Resource file list.
     {
       string result = R"(<div class="sdct_r">)";
 
-      // Handle img:example.jpg
-      QString imgTemplate( R"(<img src="bres://)" + QString::fromStdString( getId() ) + R"(/%1">)" );
+      // Resource file list.
+      // The content can be:
+      // img:pic/example.jpg	// Image file
+      // snd:apple.wav		// Sound file
+      // vdo:film.avi		// Video file
+      // att:file.bin		// Attachment file
+
+      // Extract the part after the prefix
+      static const std::map< std::string, std::string > prefixTemplates = {
```

Comment:
1. `static` variable in function only init once, and the value is unchanged across invocations.
2. std::map doesn't always improve speed, because `if-else-if or switch` that follows simple patterns will be compiled to something very fast, std::map could hinder compiler's optimizations.
3. std::map is not a continuous container like std::vector. If some data is very close to each other and fits one cache line, they will be in CPU's closest cache, fetching them one by one will be only bound by the speed of light. std::map could make things scatter around and causes more fetching, hence slower. Note that today's CPU computation speed is faster than fetching data from RAM.

Reply:
1 is error.

2,3 doesn't matter much for software of our scale in most places, but you repeatedly use this pattern, so just FYI 😅